app/vue/contexts/competition/AddCompetitionFormStepDetailsContext.js

```diff
@@ -75,6 +77,78 @@ export default class AddCompetitionFormStepDetailsContext extends BaseFuroContex
     return this.props.validationMessage
   }
 
+  /**
+   * get: initialFormValueHash
+   *
+   * @returns {PropsType['initialFormValueHash']}
+   */
+  get initialFormValueHash () {
+    return this.props.initialFormValueHash
+  }
+
+  /**
+   * get: initialTitle
+   *
+   * @returns {string | null}
+   */
+  get initialTitle () {
+    return this.initialFormValueHash
+      ?.title
+      ?? null
+  }
+
+  /**
+   * get: initialDescription
+   *
+   * @returns {string | null}
+   */
+  get initialDescription () {
+    return this.initialFormValueHash
+      ?.description
+      ?? null
+  }
+
+  /**
+   * get: initialCompetitionImageUrl
+   *
+   * @returns {string | null}
+   */
+  get initialCompetitionImageUrl () {
+    return this.initialFormValueHash
+      ?.competitionImageUrl
+      ?? null
+  }
+
+  /**
+   * get: imageSource
+   *
+   * @returns {string | null}
+   */
+  get imageSource () {
+    return this.imageSourceRef.value
+  }
+
+  /**
+   * Generate competition image URL.
+   *
+   * @returns {string}
+   */
+  generateCompetitionImageUrl () {
+    if (
+      this.initialCompetitionImageUrl === null
+      && this.imageSource === null
+    ) {
+      return FALLBACK_COMPETITION_IMAGE_URL
+    }
+
+    if (this.imageSource !== null) {
+      return this.imageSource
+    }
+
+    return this.initialCompetitionImageUrl
+      ?? FALLBACK_COMPETITION_IMAGE_URL
+  }
```

Comment:
@hecticme 

Is it the same logic? 🤔

```suggestion
    if (this.imageSource) {
      return this.imageSource
    }
  
    if (this.initialCompetitionImageUrl) {
      return this.initialCompetitionImageUrl
    }
    
    return FALLBACK_COMPETITION_IMAGE_URL
  }
```